go/private/extensions.bzl

```diff
@@ -154,7 +154,8 @@ def _default_go_sdk_name(*, module, multi_version, tag_type, index):
     # Keep the version out of the repository name if possible to prevent unnecessary rebuilds when
     # it changes.
     return "{name}_{version}_{tag_type}_{index}".format(
-        name = module.name,
+        # "main_" is not a valid module name and thus can't collide.
+        name = module.name or "main_",
```

Comment:
Should we default it `__main__` so it's the same as the default workspace name when it's not given?

Reply:
That's not possible since a repository name (which we construct here) can't start with `_`. The name of the main repository is always the empty string, `__main__` is only used for the runfiles directory without Bzlmod and will go away when WORKSPACE is removed.